apps/mobile/src/features/accounts/balances/hooks/stacks/use-stacks-pending-transactions.ts

```diff
@@ -0,0 +1,66 @@
+import { useCallback, useMemo } from 'react';
+
+import { MempoolTransaction } from '@stacks/stacks-blockchain-api-types';
+import { useQueries } from '@tanstack/react-query';
+
+import {
+  createGetAddressMempoolTransactionsQueryOptions,
+  useGetTransactionByIdListQuery,
+  useStacksClient,
+} from '@leather.io/query';
+import { isUndefined } from '@leather.io/utils';
+
+export function useGetAddressMempoolTransactionsQueries(addresses: string[]) {
+  const client = useStacksClient();
+  const queries = useQueries({
+    queries: addresses.map(address => ({
+      ...createGetAddressMempoolTransactionsQueryOptions({ address, client }),
+    })),
+  });
+  const combinedResult = useCallback(
+    () => ({
+      totalData: queries
+        .map(result => result.data)
+        .reduce((acc, result) => {
+          if (result?.results) {
+            return [...acc, ...result.results];
+          }
+          return acc;
+        }, [] as MempoolTransaction[]),
+      pending: queries.some(result => result.isPending),
```

Comment:
I will rename `pending:` here as `isPending` as it's a bit confusing. This is to indicate if querys are pending